content/learning-paths/smartphones-and-mobile/afrc/how-to-4.md

```diff
@@ -0,0 +1,172 @@
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_10BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_10BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_11BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_11BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_12BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_12BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_13BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_13BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_14BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_14BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_15BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_15BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_16BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_16BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_17BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_17BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_18BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_18BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_19BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_19BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_20BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_20BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_21BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_21BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_22BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_22BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_23BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_23BPC_BIT_EXT"},
+	     {VK_IMAGE_COMPRESSION_FIXED_RATE_24BPC_BIT_EXT, "VK_IMAGE_COMPRESSION_FIXED_RATE_24BPC_BIT_EXT"}};
+
+	std::stringstream result;
+	bool              append = false;
+	for (const auto &s : string_map)
+	{
+		if (flags & s.first)
+		{
+			if (append)
+			{
+				result << " | ";
+			}
+			result << s.second;
+			append = true;
+		}
+	}
+
+	return result.str();
+}
+```
+
+### Why does this image not support fixed-rate compression?
+
+You may find that the output of the code looks like:
+
+```output
+I/VulkanSamples: [info] Image supports VK_IMAGE_COMPRESSION_DISABLED_EXT
+I/VulkanSamples: [info] Image supports VK_IMAGE_COMPRESSION_FIXED_RATE_NONE_EXT
+```
+
+This means that the image does not support any sort of compression.
+Possible reasons are discussed in [Arm GPU Best Practices Developer Guide](https://developer.arm.com/documentation/101897/latest/Buffers-and-textures/AFRC?lang=en).
+Depending on the device, image properties like `image_create_info.format` and `image_create_info.usage` (e.g. storage images) may be incompatible with fixed-rate compression.
+
+Try using different values for these in case you find a suitable combination for your application, that supports fixed-rate compression.
+Otherwise, for best performance, try to find one that supports AFBC if possible, following the guidance found in [the guide](https://developer.arm.com/documentation/101897/latest/Buffers-and-textures/AFBC-textures-for-Vulkan?lang=en), so that the output ideally looks like this:
```

Comment:
Should it be "better performance" (as AFRC is "best performance")

also suggest: "...try to find one that at least supports AFBC if possible..."